rust/src/graph/hierarchy_queries.rs

```diff
@@ -40,11 +42,52 @@ impl Graph {
     /// Returns an iterator over the passed modules descendants.
     ///
     /// Parent modules will be yielded before their child modules.
-    pub fn get_module_descendants(&self, module: ModuleToken) -> impl Iterator<Item = &Module> {
+    pub fn get_module_descendants(&self, module: ModuleToken) -> impl ModuleIterator {
         let mut descendants = self.get_module_children(module).collect::<Vec<_>>();
         for child in descendants.clone() {
             descendants.extend(self.get_module_descendants(child.token).collect::<Vec<_>>())
         }
         descendants.into_iter()
     }
+
+    pub fn find_matching_modules(&self, expression: &ModuleExpression) -> impl ModuleIterator {
+        let interner = MODULE_NAMES.read().unwrap();
+        let modules: FxHashSet<_> = self
+            .modules
+            .values()
+            .filter(|m| expression.is_match(interner.resolve(m.interned_name).unwrap()))
+            .collect();
+        modules.into_iter()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::graph::ModuleIterator;
+    use std::collections::HashSet;
+
+    #[test]
+    fn test_find_matching_modules() {
```

Comment:
🍷  The test here is quite light, just a sanity check - we don't need to check again here how all the different types of module expressions work since that is already tested thoroughly in the dedicated unit tests (see previous commit).